### ISSUE TITLE
./configure - numerous places just say YES, NO, but not what is happening.

### ISSUE THREAD

**Issue by DuaneEllis-TI** [openthread/openthread#1600]
When running configure, it does not print the REASON for various YES/NO statements
Example output snippit from:  make -f examples/Makefile-posix

```
configure: checking whether to build tests
checking whether to build tests... yes
yes
yes
uart
yes
yes
yes
yes
no
yes
yes
no
posix
POSIX
checking whether to build tools... yes
checking for doxygen... /usr/bin/doxygen

```

**Comment by jwhui** [comment]
Thanks for raising this issue and agree that it needs to be fixed!  Would you be willing to submit a PR to address this?

**Comment by jwhui** [comment]
Resolved by #1617.
